references/common/definition_test.go

```diff
@@ -59,23 +59,29 @@ func TestDefinitionBasicFunctions(t *testing.T) {
 	if err = c.Create(context.Background(), trait); err != nil {
 		t.Fatalf("unexpected error when creating new definition with fake client: %v", err)
 	}
-	if err = def.FromCUEString(cueString + "abc"); err == nil {
+	if err = def.FromCUEString("abc:]{xa}", nil); err == nil {
+		t.Fatalf("should encounter invalid cue string but not found error")
+	}
+	if err = def.FromCUEString(cueString+"abc: {xa}", nil); err == nil {
 		t.Fatalf("should encounter invalid cue string but not found error")
 	}
 	parts := strings.Split(cueString, "template: ")
-	if err = def.FromCUEString(parts[0]); err == nil {
+	if err = def.FromCUEString(parts[0], nil); err == nil {
 		t.Fatalf("should encounter no template found error but not found error")
 	}
-	if err = def.FromCUEString("import \"strconv\"\n" + cueString); err == nil {
+	if err = def.FromCUEString("template:"+parts[1], nil); err == nil {
+		t.Fatalf("should encounter no metadata found error but not found error")
+	}
+	if err = def.FromCUEString("import \"strconv\"\n"+cueString, nil); err == nil {
```

Comment:
Just check the function not return error maybe is not enough? Can we have any way to check the parse/format result?

Reply:
Extra check is added.